core/src/main/java/org/elasticsearch/http/HttpServer.java

```diff
@@ -98,7 +100,12 @@ public void dispatchRequest(HttpRequest request, HttpChannel channel, ThreadCont
             handleFavicon(request, channel);
             return;
         }
-        restController.dispatchRequest(request, channel, threadContext);
+        try {
+            transport.addBytes(request);
+            restController.dispatchRequest(request, new ResourceHandlingHttpChannel(channel, transport), threadContext);
+        } catch (CircuitBreakingException ex) {
```

Comment:
we can make this catch throwable and remove the catch from https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/rest/RestController.java#L175